Serial.md

```diff
@@ -0,0 +1,312 @@
+  on_boot:
+    priority: -100
+    then:
+      - uart.write: "<C0.0."
+
+logger:
+  baud_rate:0
+
+api:
+  password: ""
+
+ota:
+  password: ""
+
+wifi:
+  ssid: "xxxxxxxxxxx"
+  password: "xxxxxxxxx"
+
+  ap:
+    ssid: "Desk Fallback Hotspot"
+    password: "xxxxxxxxxx"
+
+captive_portal:
+
+web_server:
+  port: 80
+
+uart:
+  id: uart_desk
+  baud_rate: 115200
+  tx_pin: D0
+  rx_pin: D1
+
+sensor:
+- platform: custom
+  lambda: |-
+    auto megadesk = new Megadesk(id(uart_desk));
+    App.register_component(megadesk);
+    return { megadesk };
+  sensors:
+  - id: megadesk_raw
+    internal: true
+    on_value:
+      then:
+        - component.update: megadesk_height_inches
+        - component.update: megadesk_height_cm
+        - component.update: megadesk_height_raw
+
+number:
+  - platform: template
+    name: "Megadesk Height (inches)"
+    id: megadesk_height_inches
+    min_value: 28.6
+    max_value: 46.75
+    step: 0.53
+    mode: slider
+    update_interval: never
+    unit_of_measurement: 'inches'
+    #NewValue = (((OldValue - OldMin) * (NewMax - NewMin)) / (OldMax - OldMin)) + NewMin
+    lambda: -|
+      return ((((id(megadesk_raw).state - 299) * (49.25 - 25.625)) / (6914 - 299)) + 25.625);
+    set_action:
+      - number.set:
+          id: megadesk_height_raw
+          value: !lambda "return int((((x - 25.625) * (6914 - 299)) / (49.25 - 25.625)) + 299);"
+  - platform: template
+    name: "Megadesk Height (cm)"
+    id: megadesk_height_cm
+    min_value: 72.644
+    max_value: 118.745
+    step: 0.53
+    mode: slider
+    update_interval: never
+    unit_of_measurement: 'cm'
+    #NewValue = (((OldValue - OldMin) * (NewMax - NewMin)) / (OldMax - OldMin)) + NewMin
+    lambda: |-
+      return ((((id(megadesk_raw).state - 299) * (125 - 65)) / (6914 - 299)) + 65);
+    set_action:
+      - number.set:
+          id: megadesk_height_raw
+          value: !lambda "return int((((x - 65) * (6640 - 299)) / (125 - 65)) + 299);"
+  - platform: template
+    name: "Megadesk Height (raw)"
+    id: megadesk_height_raw
+#    internal: true
+    min_value: 299
+    max_value: 6640
+    step: 1
+    mode: slider
+    update_interval: never
+    lambda: |-
+      return id(megadesk_raw).state;
+    set_action:
+      - uart.write: !lambda |-
+          char buf[20];
+          sprintf(buf, "<=%i,.", int(x));
+          std::string s = buf;
+          return std::vector<unsigned char>( s.begin(), s.end() );
+
+button:
+  - platform: template
+    name: "Desk Position 2"
+    on_press:
+      then:
+        - uart.write: "<L0,2."
+  - platform: template
+    name: "Desk Position 3"
+    on_press:
+      then:
+        - uart.write: "<L0,3."
+  - platform: template
+    name: "Desk Position 4"
+    on_press:
+      then:
+        - uart.write: "<L0,4."
+  - platform: template
+    name: "Desk Position 5"
+    on_press:
+      then:
+        - uart.write: "<L0,5."
+  - platform: template
+    name: "Set Minimum Desk Height"
```

Comment:
Note. this toggles the min-height limit

Reply:
And you can always query the Min height with "<R0,11."

Reply:
Changed name from "Set" to "Toggle"